src/backend/core/authentication/backends.py

```diff
@@ -49,3 +58,56 @@ def get_existing_user(self, sub, email):
             return self.UserModel.objects.get_user_by_sub_or_email(sub, email)
         except DuplicateEmailError as err:
             raise SuspiciousOperation(err.message) from err
+
+    def get_or_create_user(self, access_token, id_token, payload):
+        user = super().get_or_create_user(access_token, id_token, payload)
+        entitlement_backend = get_entitlements_backend()
+        result = entitlement_backend.can_access(user)
+        if not result["result"]:
+            raise UserCannotAccessApp(
+                result.get("message", "User does not have access to the app")
+            )
+        return user
+
+    def post_get_or_create_user(self, user, claims, is_new_user):
+        """
+        Post-processing after user creation or retrieval.
+
+        Args:
+          user (User): The user instance.
+          claims (dict): The claims dictionary.
+          is_new_user (bool): Indicates if the user was newly created.
+
+        Returns:
+        - None
+
+        """
+        claims_to_store = {
+            claim: claims.get(claim) for claim in settings.OIDC_STORE_CLAIMS
+        }
+
+        if claims_to_store:
+            user.refresh_from_db()
+            user.claims.update(claims_to_store)
+            user.save()
+
+    def create_user(self, claims):
+        """Return a newly created User instance."""
+        sub = claims.get(self.OIDC_USER_SUB_FIELD)
+        if sub is None:
+            raise SuspiciousOperation(
+                _("Claims contained no recognizable user identification")
+            )
+
+        logger.info("Creating user %s", sub)
+
+        user = self.UserModel(
+            sub=sub,
+            email=claims.get("email"),
+            full_name=claims.get("full_name"),
+            short_name=claims.get("short_name"),
+        )
+        user.set_unusable_password()
+        user.save()
+
+        return user
```

Comment:
I don't really like the idea to reimplement this method.

What about adding a `claim` key in the dictionnary returned by the `get_extra_claims` method ? This key should contain a dictionary that will be saved in the `user.claim` property. Doing this, you can remove the `create_user` and `post_get_or_create_user` methods.